File: src/models/services/AuthService.ts
import { ServiceBase } from "../../base/ServiceBase.js";
import { CredentialsModel } from "../request/CredentialsModel";
import { Response } from "../responses/Response";

export class AuthService extends ServiceBase {
  constructor() {
    super("/auth");
  }

  async signIn<T>(credentials: CredentialsModel): Promise<Response<T>> {
    return await this.post(this.url, credentials);
  }
}

File: src/models/services/AuthService.py
from src.base.ServiceBase import ServiceBase
from src.models.request.CredentialsModel import CredentialsModel

class AuthService(ServiceBase):
    def __init__(self):
        super().__init__("/auth")

    def signIn(self, credentials: CredentialsModel):
        return self.post(self.url, credentials)
